trestlebot/entrypoints/create_ssp.py

```diff
@@ -0,0 +1,143 @@
+"""
+
+import argparse
+import logging
+import pathlib
+import sys
+from typing import List
+
+from trestlebot.const import SUCCESS_EXIT_CODE
+from trestlebot.entrypoints.entrypoint_base import (
+    EntrypointBase,
+    comma_sep_to_list,
+    handle_exception,
+)
+from trestlebot.entrypoints.log import set_log_level_from_args
+from trestlebot.tasks.assemble_task import AssembleTask
+from trestlebot.tasks.authored.ssp import AuthoredSSP, SSPIndex
+from trestlebot.tasks.base_task import ModelFilter, TaskBase
+
+
+logger = logging.getLogger(__name__)
+
+
+class CreateSSPEntrypoint(EntrypointBase):
+    """Entrypoint for ssp bootstrapping."""
+
+    def __init__(self, parser: argparse.ArgumentParser) -> None:
+        """Initialize."""
+        super().__init__(parser)
+        self.setup_create_ssp_arguments()
+
+    def setup_create_ssp_arguments(self) -> None:
+        """Setup specific arguments for this entrypoint."""
+        self.parser.add_argument(
+            "--ssp-name", required=True, type=str, help="Name of SSP to create."
+        )
+        self.parser.add_argument(
+            "--profile-name",
+            required=True,
+            help="Name of profile in the trestle workspace to include in the SSP.",
+        )
+        self.parser.add_argument(
+            "--compdefs",
+            required=True,
+            type=str,
+            help="Comma-separated list of component definitions to include in the SSP",
+        )
+        self.parser.add_argument(
+            "--leveraged-ssp",
+            required=False,
+            type=str,
+            help="Provider SSP to leverage for the new SSP.",
+        )
+        self.parser.add_argument(
+            "--markdown-path",
+            required=True,
+            type=str,
+            help="Path to create markdown files in.",
+        )
+        self.parser.add_argument(
+            "--yaml-header-path",
+            required=False,
+            type=str,
+            help="Optionally set a path to a YAML file for custom SSP Markdown YAML headers.",
+        )
+        self.parser.add_argument(
+            "--version",
+            required=False,
+            type=str,
+            help="Optionally set the SSP version.",
+        )
+        self.parser.add_argument(
+            "--ssp-index-path",
+            required=False,
+            type=str,
+            default="ssp-index.json",
+            help="Optionally set the path to the SSP index file.",
+        )
+
+    def run(self, args: argparse.Namespace) -> None:
+        """Run the entrypoint."""
+        exit_code: int = SUCCESS_EXIT_CODE
+        try:
+            set_log_level_from_args(args)
+
+            # If the ssp index file does not exist, create it.
+            ssp_index_path = pathlib.Path(args.ssp_index_path)
+            if not ssp_index_path.exists():
+                # Create a parent directory
+                ssp_index_path.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
minor: unclear if **exist_ok=True** required here

Reply:
Ah. Good catch. Not needed.